worlds/witness/__init__.py

```diff
@@ -150,27 +133,107 @@ def generate_basic(self):
             location_obj = self.multiworld.get_location(event_location, self.player)
             location_obj.place_locked_item(item_obj)
 
-        self.multiworld.itempool += pool
+        # Find out how much empty space there is for junk items. -1 for the "Town Pet the Dog" check
+        itempool_difference = len(self.locat.CHECK_LOCATION_TABLE) - len(self.locat.EVENT_LOCATION_TABLE) - 1
+        itempool_difference -= len(pool)
 
-    def pre_fill(self):
-        # Put good item on first check if there are any of the designated "good items" in the pool
+        # Place two locked items: Good symbol on Tutorial Gate Open, and a Puzzle Skip on "Town Pet the Dog"
         good_items_in_the_game = []
+        plandoed_items = set()
+
+        for v in self.multiworld.plando_items[self.player]:
+            if v["from_pool"]:
+                plandoed_items.update({self.items_by_name[i] for i in v["items"].keys()})
+            else:
+                itempool_difference -= 1
```

Comment:
Hm, I can identify see 3 things wrong with this :S

1. "from_pool" is optional, defaulting to `true` - i.e. use `if (v.get("from_pool", True)` instead
2. there are two modes: `item` and `items`
3. there is also an optional count. you can place "2 out of those 3" for example

Reply:
Ah okay. Honestly, I can get rid of the "else" again. I'm "fixing" an issue that exists across all games and is fundamental to from_pool: false.

Reply:
What do you mean by 2.? can ["items"] break?

Reply:
I made a change, do you think that could've done the trick? I now check both "item" and "items" and I check whether the item actually exists, and I use defaults for values that don't exist...